### PR TITLE
Convert codelens-handler.js to typescript

### PR DESCRIPTION
**Pull request by MarkusJx**
This will convert ``static/codelens-handler.js`` to typescript (see #2981)

## Changes made
* Converted everything to typescript
* Added custom types for some inputs
* Updated ``static/tsconfig.json``
* Replaced ``var`` with ``let`` and ``const`` wherever possible

## Notes
* The whole module can be built with the changes made and
* The tests do pass

### CODE REVIEW

File: static/codelens-handler.ts [hunk 1]
```diff
@@ -24,14 +24,20 @@
 
 'use strict';
```
Comment (supergrecko) [inline static/codelens-handler.ts:25]: Unnecessary 'use strict'

**Review by supergrecko — APPROVED** [review]

**Comment by MarkusJx** [comment]
Done.

**Review by mattgodbolt — APPROVED** [review]
Thanks so much!

**Comment by mattgodbolt** [comment]
Now running on the live site, thanks!

**Pull request merged**